### PR TITLE
CHE-6914: Move Java Debugger Location Conversion to jdt.ls extension

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
Move Java Debugger Location Conversion to jdt.ls extension

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/6914

### CODE REVIEW

**Review by svor — APPROVED** [review]

File: plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java [hunk 1]
```diff
@@ -302,4 +309,49 @@ private void fixLocation(ExtendedSymbolInformation symbol) {
...
+    CompletableFuture<Object> result =
+        getLanguageServer()
+            .getWorkspaceService()
+            .executeCommand(
+                new ExecuteCommandParams(
+                    Commands.IDENTIFY_FQN_IN_RESOURCE,
+                    ImmutableList.of(prefixURI(filePath), String.valueOf(lineNumber))));
+
+    try {
+      return (String) result.get();
```
Comment (mmorhun) [inline plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java]: Timeout?

**Review by mmorhun — APPROVED** [review]

File: agents/ls-java/src/main/resources/installers/1.0.1/org.eclipse.che.ls.java.script.sh [hunk 2]
```diff
@@ -135,4 +135,4 @@ curl -sL ${DOWNLOAD_AGENT_BINARIES_URI} | tar xzf - -C ${LS_DIR}
 
 touch ${LS_LAUNCHER}
 chmod +x ${LS_LAUNCHER}
-echo "java -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -noverify -Xmx1G -jar ${LS_DIR}/plugins/org.eclipse.equinox.launcher_1.4.0.v20161219-1356.jar -configuration ./config_linux -data ${LS_DIR}/data" > ${LS_LAUNCHER}
+echo "java -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -noverify -Xmx1G -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=4410 -jar ${LS_DIR}/plugins/org.eclipse.equinox.launcher_1.4.0.v20161219-1356.jar -configuration ./config_linux -data ${LS_DIR}/data" > ${LS_LAUNCHER}
```
[the remark this answers is not in the document]
Comment (tolusha) [reply agents/ls-java/src/main/resources/installers/1.0.1/org.eclipse.che.ls.java.script.sh]: For some time.

File: plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/debug/FileResourceLocationHandler.java [hunk 3]
```diff
@@ -31,8 +31,8 @@
 @Singleton
 public class FileResourceLocationHandler implements DebuggerLocationHandler {
 
-  private final EditorAgent editorAgent;
-  private final AppContext appContext;
+  protected final EditorAgent editorAgent;
```
Comment (tsmaeder) [inline plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/debug/FileResourceLocationHandler.java:34]: I don't see a reason to make these fields protected.

File: plugins/plugin-java-debugger/che-plugin-java-debugger-server/src/test/resources/testclasses/ThreadDumpTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
+/*
```
Comment (tsmaeder) [inline plugins/plugin-java-debugger/che-plugin-java-debugger-server/src/test/resources/testclasses/ThreadDumpTest.java:1]: This looks more like an experiment someone did than a test. Do we need it?

**Review by tsmaeder — CHANGES_REQUESTED** [review]

File: plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/debug/FileResourceLocationHandler.java (hunk 3 above)
Comment (tolusha) [reply plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/debug/FileResourceLocationHandler.java:34]: To be visible in children classes.

File: plugins/plugin-java-debugger/che-plugin-java-debugger-server/src/test/resources/testclasses/ThreadDumpTest.java (hunk 4 above)
Comment (tolusha) [reply plugins/plugin-java-debugger/che-plugin-java-debugger-server/src/test/resources/testclasses/ThreadDumpTest.java:1]: Test classes for java debugger server side tests.
We do need them.

File: plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/debug/FileResourceLocationHandler.java (hunk 3 above)
Comment (tsmaeder) [reply plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/debug/FileResourceLocationHandler.java:34]: Yes, but why is that necessary?

Comment (tolusha) [reply plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/debug/FileResourceLocationHandler.java:34]: It is used there.

**Review by vparfonov — APPROVED** [review]

**Review by tsmaeder — APPROVED** [review]

**Pull request merged**